spec/models/profile_spec.rb

```diff
@@ -305,32 +538,62 @@
 
   describe '#deactivate' do
     let(:deactivation_reason) { :password_reset }
-    let(:profile) do
+
+    it 'sets active flag to false and assigns deactivation_reason' do
       profile = create(:profile, :active, user: user)
+
+      expect(profile.activated_at).to be_nil
+      expect(profile.active).to eq(true) # to change
+      expect(profile.deactivation_reason).to be_nil # to change
+      expect(profile.fraud_review_pending?).to eq(false)
+      expect(profile.gpo_verification_pending_at).to be_nil
+      expect(profile.initiating_service_provider).to be_nil
+      expect(profile.verified_at).to be_nil
+
       profile.deactivate(deactivation_reason)
-      profile
-    end
 
-    it 'sets active flag to false' do
+      expect(profile.activated_at).to be_nil
+      expect(profile.active).to eq(false) # changed
+      expect(profile.deactivation_reason).to eq('password_reset') # changed
+      expect(profile.fraud_review_pending?).to eq(false)
+      expect(profile.gpo_verification_pending_at).to be_nil
+      expect(profile.initiating_service_provider).to be_nil
+      expect(profile.verified_at).to be_nil
+
       expect(profile).to_not be_active
       expect(profile).to be_password_reset
     end
   end
 
   describe '#remove_gpo_deactivation_reason' do
     it 'removes the gpo_verification_pending_at deactivation reason' do
-      profile = create(
-        :profile, user: user, active: false,
-                  gpo_verification_pending_at: 1.day.ago
-      )
+      profile = create(:profile, :verify_by_mail_pending)
+
+      expect(profile.activated_at).to be_nil
+      expect(profile.active).to eq(false)
+      expect(profile.deactivation_reason).to be_nil
+      expect(profile.fraud_review_pending?).to eq(false)
+      expect(profile.gpo_verification_pending_at).to be_present # to change
+      expect(profile.initiating_service_provider).to be_nil
+      expect(profile.verified_at).to be_nil
+
       profile.remove_gpo_deactivation_reason
 
+      expect(profile.activated_at).to be_nil
+      expect(profile.active).to eq(false)
+      expect(profile.deactivation_reason).to be_nil
+      expect(profile.fraud_review_pending?).to eq(false)
+      expect(profile.gpo_verification_pending_at).to be_nil # changed
+      expect(profile.initiating_service_provider).to be_nil
+      expect(profile.verified_at).to be_nil
+
       expect(profile.gpo_verification_pending?).to be(false)
     end
   end
 
   describe '#activate_after_password_reset' do
-    it 'activates a profile after password reset' do
+    it 'activates a non-verified, non-active, non-activated profile after password reset' do
+      # TODO: this profile scenario shouldn't be possible
```

Comment:
As discussed, this scenario should expect a RuntimeError, and the code should raise.